solr/core/src/java/org/apache/solr/handler/admin/api/ReplicationAPIBase.java

```diff
@@ -38,10 +68,140 @@ public ReplicationAPIBase(
   }
 
   protected CoreReplicationAPI.IndexVersionResponse doFetchIndexVersion() throws IOException {
+    ReplicationHandler replicationHandler =
+        (ReplicationHandler) solrCore.getRequestHandler(ReplicationHandler.PATH);
+    return replicationHandler.getIndexVersionResponse();
+  }
 
+  protected NamedList<Object> doFetchFiles(long generation) {
     ReplicationHandler replicationHandler =
         (ReplicationHandler) solrCore.getRequestHandler(ReplicationHandler.PATH);
+    return getFileList(generation, replicationHandler);
+  }
 
-    return replicationHandler.getIndexVersionResponse();
+  protected NamedList<Object> getFileList(long generation, ReplicationHandler replicationHandler) {
+    final IndexDeletionPolicyWrapper delPol = solrCore.getDeletionPolicy();
+    final NamedList<Object> filesResponse = new NamedList<>();
+
+    IndexCommit commit = null;
+    try {
+      if (generation == -1) {
```

Comment:
[Q] Mostly thinking out loud here, but I wonder whether we should just fall back on "-1"
 as a default.  (Right now we require users to provide an explicit generation value).
 
 Probably not something to handle in this PR regardless, just wanted to jot down the thought somewhere...

Reply:
This makes sense to me and like that idea. I could try changing it myself in this PR but if you think this requires more discussion or would like to do it yourself in another PR I'll leave as is.

Reply:
Hmmm I agree it makes less sense to put generation in the path if we decide to support a default generation. There doesn't seem to be a clean way to generation in the path while also defaulting so if I had to choose, I think there is more value in generation being in the query parameter and being able to default generation than the other way around. 

I'll modify the API with generation in the query parameter.

Reply:
API updated with generation now as a query parameter